tree/tree/test/TTreeRegressions.cxx

```diff
@@ -230,3 +232,62 @@ TEST(TTreeRegressions, EmptyLeafObject)
    TLeafObject tlo;
    EXPECT_EQ(tlo.GetObject(), nullptr);
 }
+
+// https://its.cern.ch/jira/browse/ROOT-6741
+class MySubClass {
+public:
+   MySubClass(int Id = 0, double X = 0) : id(Id), x(X) {}
+   virtual ~MySubClass() {}
+   int id;
+   double x;
+};
+class MyClass {
+public:
+   virtual ~MyClass() {}
+   std::vector<MySubClass> sub;
+   void Push(MySubClass msc) { sub.push_back(msc); }
+   MySubClass *Get(int id)
+   {
+      for (size_t i = 0; i < sub.size(); ++i)
+         if (sub[i].id == id)
+            return &sub[i];
+      return 0;
+   }
+};
+TEST(TTreeRegressions, TTreeFormulaMemberIndex)
+{
+   gInterpreter->Declare("class MySubClass { public: MySubClass(int Id = 0, double X = 0) : id(Id), x(X) {} virtual ~MySubClass() {} int id; double x; };");
```

Comment:
The code duplication here (and the next line) is sub-optimal.  Would it be possible to use a (private) header file instead?

Reply:
I guess so, (though I copy-pasted the structure from the test by Enrico in the same file above), so not sure if it's worth to change all nor to change just this one (inconsistency) ? Also if someone deletes the test later on he might forget to delete the header file, this way it's more self-contained?

Maybe a trick with a preprocessor stringification could be a good compromise to avoid duplication? (not fully sure if it's possible)